app/clients/zendesk_sell_client.py

```diff
@@ -0,0 +1,94 @@
+import json
+import requests
+
+from typing import Dict, List, Union
+
+from flask import current_app
+
+from app.authentication.bearer_auth import BearerAuth
+from app.user.contact_request import ContactRequest
+
+__all__ = [
+    'ZenDeskSellClient'
+]
+
+
+class ZenDeskSellClient(object):
+
+    def _generate_lead_data(self) -> Dict[str, Union[str, List[str], Dict]]:
+
+        # The API and field definitions are defined here: https://developers.getbase.com/docs/rest/reference/leads
+
+        # validation based upon api mandatory fields
+        assert len(self.contact.name) or len(self.contact.department_org_name), 'Name or Org are mandatory field'
+
+        recipients = {
+            'internal': 'Colleagues within your department (internal)',
+            'external': 'Partners from other organizations (external)',
+            'public': 'Public'
+        }
+        # FIXME: consider migrating to pypi/nameparser for proper name parsing to handle cases like:
+        # 'Dr. Juan Q. Xavier de la Vega III (Doc Vega)'
+        name_tokenised = self.contact.name.split()
+        return {
+            'data': {
+                'last_name': name_tokenised[-1],
+                'first_name': " ".join(name_tokenised[:-1]) if len(name_tokenised) > 1 else '',
+                'organization_name': self.contact.department_org_name,
+                'email': self.contact.email_address,
+                'description': f'Program: {self.contact.program_service_name}\n{self.contact.main_use_case}: '
+                               f'{self.contact.main_use_case_details}',
+                'tags': [self.contact.support_type, self.contact.language],
+                'status': 'New',
+                'custom_fields': {
+                    'Product': ['Notify'],
+                    'Source': 'Demo request form',
+                    'Intended recipients': recipients[self.contact.intended_recipients]
+                    if self.contact.intended_recipients in recipients else 'No value'
```

Comment:
This field is required on the admin https://github.com/cds-snc/notification-admin/blob/41e2098669c9b70778f7a48864a2bea78dd48cfe/app/main/forms.py#L947-L954

Reply:
That's true however this is not a requirement for the zendesk sell integration. We keep the code here loosely coupled so that if anything changes within the notification-admin we'd be able to handle it properly when we are sending to zendesk sell.